Bursts/matlab/getBinnedBurstInfo.m

```diff
@@ -1,48 +1,53 @@
 %   - 0.5 spikes/sec/neuron as a threshold for burst detection
 %   - for 10ms (0.01s) bin, burst threshold = 0.5x0.01 = 0.005
-b_threash = 0.005;                              % burst threashold
-b_bins = find(SH10ms >= b_threash);             % find bins > threshold
-b_boundary = find(diff(b_bins) > 1);            % find burst boundaries
-n_bursts = length(b_boundary)+1;                % total bursts
-b_boundary = [0; b_boundary];                   % boundary condition
-b_boundary = [b_boundary; length(b_bins)];      % boundary condition
-prev_peak = 0;
+adjustedBurstThreshold = 0.005;                                                    % burst threashold adjusted for 10ms Bin
+binsAboveThreshold = find(spikesPerNeuronPer10msBin >= adjustedBurstThreshold);    % find index of bins above threshold
+% find burst boundaries, if the bins above the threshold is not next to each other 
+% then get their indexes
+burstBoundaries = find(diff(binsAboveThreshold) > 1);               % contain indexes of bins above the threshold
+nBursts = length(burstBoundaries)+1;                                % number of bursts detected
+burstBoundaries = [0; burstBoundaries];                             % boundary condition
+burstBoundaries = [burstBoundaries; length(binsAboveThreshold)];    % boundary condition
+previousPeak = 0;
+
 % Output file
-fid = fopen([h5dir '/binnedBurstInfo.csv'], 'w') ;         
+fid = fopen([h5dir '/allBinnedBurstInfo.csv'], 'w') ;         
 fprintf(fid, ['ID,startBin#,endBin#,width(bins),totalSpikeCount,'... 
               'peakBin,peakHeight(spikes),Interval(bins)\n']);
-for i = 1:n_bursts
-    b.id = i;                                       % burst ID
-    b.start = b_bins(b_boundary(i)+1);              % start bin number
-    b.ended = b_bins(b_boundary(i+1));              % end bin number
-    b.width = (b.ended-b.start)+1;                  % burst width in bins 
-    b.count = sum(SH(b.start:b.ended));             % total spike count
-    [b.height, temp] = max(SH(b.start:b.ended));    % height of the peak
-    b.peak = b.start+temp-1;            % peak bin (highest spike count)                
-    b.interval = b.peak-prev_peak;      % burst interval in bins
+for iBurst = 1:nBursts
+    Burst.startBinNum = binsAboveThreshold(burstBoundaries(iBurst)+1);      % start bin number
+    Burst.endBinNum = binsAboveThreshold(burstBoundaries(iBurst+1));        % end bin number
+    Burst.width = (Burst.endBinNum - Burst.startBinNum)+1;                  % burst width in bins 
+    Burst.totalSpikeCount = sum(spikesPerBin(Burst.startBinNum:Burst.endBinNum));       % total spike totalSpikeCount
+    [Burst.peakHeight, temp] = max(spikesPerBin(Burst.startBinNum:Burst.endBinNum));    % height of the peak
+    Burst.peakBin = Burst.startBinNum+temp-1;                               % peak bin (highest spike Count)                
+    Burst.interval = Burst.peakBin-previousPeak;                            % burst interval in bins
```

Comment:
There is no reason to use struct here over regular vector array. I thought it didn't need to changed because this function already runs under half a second. Struct is slower so I have changed this to regular array.